src/tools/cephfs_mirror/PeerReplayer.cc

```diff
@@ -1167,6 +1167,22 @@ int PeerReplayer::do_synchronize(const std::string &dir_root, const Snapshot &cu
   }
 
   struct ceph_statx tstx;
+
+  // sync the mode of the remote dir_root with that of the local dir_root
+  r = ceph_statx(m_local_mount, dir_root.c_str(), &tstx, CEPH_STATX_MODE,
+		 AT_STATX_DONT_SYNC | AT_SYMLINK_NOFOLLOW);
+  if (r < 0) {
+    derr << ": failed to fetch stat for local dir root: "
+	 << cpp_strerror(r) << dendl;
+    return r;
+  }
+  r = ceph_fchmod(m_remote_mount, fh.r_fd_dir_root, tstx.stx_mode);
```

Comment:
Why not do this when creating the directory structure in `PeerReplayer::try_lock_directory()`?

Reply:
mostly because this has nothing to do with directory locking

Reply:
True, but that's when the remote directory is created, and it makes sense to fix the perms at that point.

Reply:
revised the code